packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -27,7 +27,28 @@ import {
 	VisualNodeKind,
 	type VisualValueNode,
 	type VisualTreeNode,
+	type FluidHandleNode,
+	type UnknownObjectNode,
 } from "./VisualTree.js";
+import type { VisualizeChildData } from "./DataVisualization.js";
+
+/**
+ * Returns VisualNodeKind that is compatible to {@link FluidObjectNode} based on the `visualTree`'s node kind.
+ */
+export function determineNodeKind(nodeKind: VisualNodeKind): VisualNodeKind {
```

Comment:
Since this logic is very specific to the `visualizeSharedTree` function, and isn't really reusable in any other context, I would recommend inlining it in that function.